kedro/io/data_catalog.py

```diff
@@ -567,16 +642,47 @@ def list(self, regex_search: str | None = None) -> list[str]:
             ) from exc
         return [dset_name for dset_name in self._data_sets if pattern.search(dset_name)]
 
+    def exists_in_catalog_config(self, dataset_name: str) -> bool:
```

Comment:
Is this the cache that add matched pattern into catalog? If so why do we need to keep it separately instead of just creating the entry in catalog?

Reply:
This fn just adds the dataset name -> pattern in the `self._pattern_matches_cache` does not create an entry in the catalog. This is because this function is also called in the runner to check for existence of datasets from the pipeline in the catalog.  The actual adding of a dataset to the catalog happens in `self._get_dataset()`
This is to avoid doing the pattern matching multiple times when it's called inside a Session run but the actual resolving and adding is still at `_get_dataset` time to allow for DataCatalog standalone usage

Reply:
This isn't a good name for a public method, it's way too descriptive. I would suggest that you change it to `def __contains__(self, dataset_name):`. This will allow us to do a test like this:
`if dataset_name in catalog:...`, which is way more intuitive and nicer than `catalog.exists_in_catalog_config(dataset_name)`.